### ISSUE TITLE
[750$] Advanced Search Page

### ISSUE THREAD

**Issue by ItsANameToo** [ArkEcosystem/explorer#703]
**About This Project**

Core contains various endpoints for searching, e.g.:

- Blocks: https://docs.ark.io/api/public/v2/blocks.html#search-all-blocks
- Transactions: https://docs.ark.io/api/public/v2/transactions.html#search-for-transactions
- Wallets: https://docs.ark.io/api/public/v2/wallets.html#search-all-wallets

For the explorer, we want to see a dedicated search page that utilizes what these endpoints bring. This means there should be options to search for blocks, transactions, and wallets based on the parameters that are allowed by the endpoints. Delegate search is not considered in this issue, as that is a better fit for the existing search field at the top of the explorer page as defined in #665.

> The difficulty level of this bounty is estimated as **intermediate**. The reward for completion of this project is set at **$750** (paid in ARK Ѧ).

_Requirements for completion:_
- Dedicated search page to search through blocks, transactions, and wallets
- Have proper inputs for the different types of search parameters so it’s easy for users to use them.
- Show a list of matching results that link to the corresponding explorer pages
- Proper tests that cover the newly added code

_Necessary Skills:_
- [TypeScript](https://www.typescriptlang.org/)
- [Vue.JS](https://vuejs.org)
- [Jest](https://jestjs.io/)
- [Cypress](https://www.cypress.io)

**Applying For This Project**

To apply for this project, please read more about the process below before leaving a comment with the following:

1. A brief proposal relating to the project (e.g. Your experience, how you wish to approach the project, estimated time to completion etc.)
2. Your name and a contact email address (or another form of contact, eg. Slack username) 

**The Process**

If your application for the project is successful we'll contact you using the details you provided, assign the project to you and mark it as in progress. You'll be required to check in regularly with the ARK team to discuss progress and development. We'll also use these updates to ensure that the PR is being completed to a satisfactory standard. If we determine that the project isn't being completed to a satisfactory standard and that this cannot be achieved with minor amends or within a satisfactory timeframe, we hold the right to dismiss the PR and reopen the project to other applicants.

To deliver the project your PR must be submitted to the respective repo. ARK repo maintainers will then leave comments and request amends if necessary. All amends requested by maintainers must be resolved before completion, and all projects must comply with our code standards and contribution guidelines (found at: https://docs.ark.io/guidebook/contribution-guidelines/). After you've submitted your project, we'll review against the brief to ensure it meets the criteria and deliverables.

Once your project is approved as complete a member of the ARK team will follow up to discuss payment. You'll need to submit an invoice and payment will be issued in (Ѧ) ARK (equivalent to the USD reward value assigned to the project), sent to an address of your choosing. We may need to request additional information regarding your identity in order to comply with our own internal Know Your Customer (KYC) guidelines.

_If you're unsuccessful in applying for this particular project, please consider applying for others. Thanks for your interest and good luck with your application!_

**Comment by dated** [comment]
I would like to submit my application for this bounty as I have extensive experience both with the Explorer application itself as with the Core API, having found many bugs and issues which were partially patched by me. Development on this task has been started already, but has been put on hold to avoid unneccessary refactoring / changes once official designs for this feature are available.

As laid out above, further development from my side would require official designs to follow. The completion of this project should not take more than 10 days - including test cases.

The requirement for extending the test suite will be that the migration to cypress (#701) is completed and merged by that point.

Edgar Goetzendorff / @dated [slack]

**Comment by dated** [comment]
Will @olejegcord be providing the designs for this task, @ItsANameToo?
